src/config.h

```diff
@@ -58,6 +58,7 @@ struct vpn_config {
 	char		username[FIELD_SIZE];
 	char		password[FIELD_SIZE];
 	char		cookie[COOKIE_SIZE + 1];
+	char            *realm;
```

Comment:
It's better to copy the realm string into the config struct, rather than keeping a pointer to a memory area that can change. You should use:

```c
	char		realm[FIELD_SIZE];
```